### PR TITLE
feat: ✨ adds `QuoteTweet` component

### PR DESCRIPTION
**Pull request by Roaa94**
## Status

**READY**

## Description

* Adds the widget `QuotedTweet` for tweets quoted inside other tweets (`ExpandedTweet` & `CollapsedTweet`) with thumbnail and large media use cases
* Adds a `Quote Tweet` use case to `ExpandedTweet` & `CollapsedTweet` components with a `QuoteTweet` widget included.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: app_widgetbook/lib/core/tweet/collapsed_tweet.dart [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import 'package:app_widgetbook/dummy_data/dummy_media.dart';
+import 'package:app_widgetbook/dummy_data/dummy_tweets.dart';
 import 'package:app_widgetbook/dummy_data/dummy_users.dart';
+import 'package:app_widgetbook/utils.dart';
 import 'package:core/core.dart';
-import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
```
Comment (jenshor) [inline app_widgetbook/lib/core/tweet/collapsed_tweet.dart:5]: image
do you also see the rectangles?

File: packages/core/lib/src/models/public_metrics.dart [hunk 2]
```diff
@@ -19,4 +19,9 @@ class PublicMetrics {
 
   /// Tweet replies count
   final int replies;
+
+  /// Whether all metrics except replies are zero
+  bool get zero {
+    return retweets == 0 && quoteTweets == 0 && likes == 0;
+  }
```
Comment (jenshor) [inline packages/core/lib/src/models/public_metrics.dart]: better name might be `isZero`. Otherwise, I'd expect some sort of `num` to be returned.

**Review by jenshor — APPROVED** [review]
Found some minor stuff. We can fix it in #22.

File: app_widgetbook/lib/core/tweet/collapsed_tweet.dart (hunk 1 above)
Comment (jenshor) [reply app_widgetbook/lib/core/tweet/collapsed_tweet.dart:5]: This is happening in the Cloud not sure if it also appears locally.

File: packages/core/lib/src/models/public_metrics.dart (hunk 2 above)
Comment (Roaa94) [reply packages/core/lib/src/models/public_metrics.dart]: Agreed 👍

File: app_widgetbook/lib/core/tweet/collapsed_tweet.dart (hunk 1 above)
Comment (Roaa94) [reply app_widgetbook/lib/core/tweet/collapsed_tweet.dart:5]: Oh yes I noticed that. This seems to only happen on web. I will look into it in the next PR

**Pull request merged**